src/main/java/org/opensearch/knn/index/query/rescore/RescoreContext.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.query.rescore;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.opensearch.common.ValidationException;
+
+@Slf4j
+@Getter
+@AllArgsConstructor
+@Builder
+@EqualsAndHashCode
+public final class RescoreContext {
+
+    public static final float DEFAULT_OVERSAMPLE_FACTOR = 1.0f;
+    public static final float MAX_OVERSAMPLE_FACTOR = 5.0f;
+    public static final float MIN_OVERSAMPLE_FACTOR = 0.0f;
+
+    @Builder.Default
+    private float oversampleFactor = DEFAULT_OVERSAMPLE_FACTOR;
+
+    /**
+     * Validate the rescore context
+     *
+     * @return ValidationException if validation fails, null otherwise
+     */
+    public ValidationException validate() {
```

Comment:
Out of curiosity, there are places in the code base where there is a similar pattern, The POJO is coupled with validation. Ideally these should just have getters and setters, validations should be a separate concern, we ideally wouldn't want to have access to any logic apart from getters and setters to wherever this object is being passed around

Thoughts?

Reply:
I think that makes sense. Let me update to move it to parser similar to method params

Reply:
done